server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -821,6 +825,30 @@ ClusterNode getAnyNodePreferringLeaders() {
             return getAnyNode();
         }
 
+        class MockPersistedState extends InMemoryPersistedState {
+            MockPersistedState(long term, ClusterState acceptedState) {
+                super(term, acceptedState);
+            }
+
+            private void possiblyFail(String description) {
+                if (disruptStorage && rarely()) {
+                    throw new CoordinationStateRejectedException("simulated IO exception [" + description + ']');
```

Comment:
Can we throw other exception types here? Can you add TODO that we should check whether the same needs to be done for IOException?